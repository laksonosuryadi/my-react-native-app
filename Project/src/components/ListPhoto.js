import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Text,
  View,
  ScrollView,
  ListView,
  Image,
  RefreshControl,
  TouchableHighlight
} from 'react-native';

import { Col, Grid, Spinner, Footer, FooterTab, Button, Icon } from 'native-base';

import PhotoItem from './PhotoItem'
import { fetchPhoto, fetchMyData } from '../actions'

class ListPhoto extends Component {
  constructor(props) {
    super(props)
    this.state = {
      refreshing: false,
    };
  }

  static navigationOptions = { title : '700px'}

  componentDidMount() {
    this.props.fetchPhoto()
    this.props.fetchMyData()
  }

  afterRefresh() {
    this.setState({refreshing: false})
    this.props.fetchPhoto()
  }

  _onRefresh() {
    this.setState({refreshing: true});
    this.afterRefresh()
  }

  render () {
    const { navigate } = this.props.navigation
    return (
      <View style={styles.container}>
        <ScrollView
          style={styles.content}
          refreshControl={
           <RefreshControl
             refreshing={this.state.refreshing}
             onRefresh={this._onRefresh.bind(this)}
           />
        }>
          { this.props.photos.length !== 0 &&
            <View style={{flex:1, alignItems:'center', justifyContent:'center'}}>
              <Text style={{fontSize:25, marginTop:10, marginBottom:10}}>20 Popular Photos</Text>
            </View>
          }

          { this.props.photos.length === 0 && <Spinner color='grey' style={{marginTop:200}}/>}

          <View style={styles.grid}>
              {
                this.props.photos.map(photo => (
                  <PhotoItem key={photo.id} photo={photo} navigation={this.props.navigation} />
                  )
                )
              }
          </View>

          { this.props.photos.length !== 0 &&
            <Footer>
              <FooterTab>
                <Button onPress={() => navigate('ListPhoto')}>
                  <Icon name="home" />
                </Button>
                <Button>
                  <Icon name="camera" />
                </Button>
                <Button onPress={() => navigate('About', {data: this.props.user})}>
                  <Icon name="person" />
                </Button>
              </FooterTab>
            </Footer>
          }
        </ScrollView>
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  content: {
    flex: 1,
    width: '100%'
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
    marginBottom: 20
  },
  grid: {
    flexDirection: 'row',
    flexWrap:'wrap'
  },
  loading: {
    marginTop: 100,
    fontSize: 30,
    justifyContent: 'center',
    alignSelf: 'center'
  },
  stats: {
    marginLeft: 10,
    fontSize: 20,
    fontWeight: 'bold',
    color: 'black'
  }
}

const mapStateToProps = (state) => ({
  photos: state.data.photos,
  user: state.data.user
})

const mapDispatchToProps = dispatch => ({
  fetchPhoto: () => dispatch(fetchPhoto()),
  fetchMyData: () => dispatch(fetchMyData())
})

export default connect(mapStateToProps, mapDispatchToProps)(ListPhoto);
