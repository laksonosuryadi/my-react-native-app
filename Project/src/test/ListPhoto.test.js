import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import ListPhoto from '../components/ListPhoto'
import store from '../store';

it('render ListPhoto successfully', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ListPhoto photo={{}} navigation={{}}/>
    </Provider>
  )
  expect(tree).toMatchSnapshot();
})
