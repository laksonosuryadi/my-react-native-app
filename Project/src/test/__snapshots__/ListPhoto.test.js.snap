// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ListPhoto successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://avatars0.githubusercontent.com/u/25143119?v=3&u=23b4af0333afa56e2575ed1b72ea469fa733108c&s=400",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 80,
                "margin": 10,
                "width": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "marginTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginLeft": 10,
              }
            }
          >
            1234
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            posts
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "marginTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginLeft": 10,
              }
            }
          >
            999
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            followers
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "marginTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginLeft": 10,
              }
            }
          >
            888
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            following
          </Text>
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          Laksono Suryadi
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "marginBottom": 4,
              "marginLeft": 10,
            }
          }
        >
          A photography enthusiast
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 12,
              "marginBottom": 20,
              "marginLeft": 10,
            }
          }
        >
          Followed by
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontWeight": "bold",
              }
            }
          >
             Anthony Juan, Andra Satria, + 10 more
          </Text>
        </Text>
      </View>
      <ActivityIndicator
        animating={true}
        color="#45D56E"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "height": 80,
          }
        }
        virtual={undefined}
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
